src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/aarch64/aapcs/AapcsLinker.java

```diff
@@ -46,8 +47,8 @@
  * ABI implementation based on ARM document "Procedure Call Standard for
  * the ARM 64-bit Architecture".
  */
-public class AArch64Linker implements CLinker {
-    private static AArch64Linker instance;
+public class AapcsLinker implements CLinker {
```

Comment:
I see why it's called like that - but this is really either AArch64Linker or Arm64Linker. E.g. the platform standard one.

Reply:
How about StandardAArch64Linker? Otherwise MacOsAArch64Linker sounds like it should be a specialisation of AArch64Linker.

Reply:
Or LinuxAArch64Linker if we're going to rename the CABI enum AArch64 -> LinuxAArch64? But this would also be used on other OSs that use the Arm ABI spec as-is (e.g. *BSD).